### PR TITLE
mgr/dashboard: Update translations

### PR DESCRIPTION
**Pull request by Exotelis**
This pr is going to update the following languages:

- Czech
- German
- Spanish
- French
- Indonesian
- Italian
- Japanese
- Polish
- Portuguese (Brazil)
- Chinese (China)
- Chinese (Taiwan)

I also added Korean as new language to the dashboard and added a messages.en-US.xlf which is basically the source language. But when exporting the source languages it will generate a messages.xlf which we do not track in the repo. But since we moved from JIT to AOT we need to have an english source file for building the english package.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by Exotelis** [comment]
jenkins test dashboard

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by Exotelis** [comment]
In the standup meeting I phrased it a little bit wrong, I did not explicitly add the messages.en-US.xlf to the repository, it has been there already. But I did add en-US to the languages that should be updated when synchronizing with transifex ([see](https://github.com/ceph/ceph/pull/31435/files#diff-adfd988a555fd872e0488da1c1ef7f7eR5)). 

Since we moved from JIT to AOT the messages.en-US-xlf became more important. The problem is how we build the frontend now.  I'd like to highlight those 2 lines:

`"build-i18n": "export _locale=${LOCALE:-$npm_package_config_locale}; [ -f src/locale/messages.${_locale}.xlf ] && npm run build -- --outputPath=dist/${_locale} --i18nFile=src/locale/messages.${_locale}.xlf --i18nLocale=${_locale}",`

`"build:en-US": "LOCALE=en-US npm run build-i18n",`

As you can see, when running the build:en-US script, we set the locale to en-US. Then we call the build-i18n script. In this script the locale will be used to find the correct xlf file for the package that should be built (`src/locale/messages.${_locale}.xlf `).

This alone is not a big problem, but it can cause some.

Imagine one rephrases the text of the menu item Dashboard to Dashb. Also imagine that a test exists which checks if a menu item with the text Dashboard exists. To prevent any failures the person who changed the text to Dashb. would also update the test. But against all assumptions the test would still fail, because the messages.en-US.xlf hasn't been updated. To do this, one would have to run the i18n script and update the translation files, because the test would have been updated but the text would remain Dashboard until one updated the translations. However this should only be a problem in e2e tests, no?

I see multiple solutions for this. Haven't tested any of them but they should work:
- Don't use any xlf file when building the source language package. An xlf file should not be needed, since the orignal strings are all available in the TypeScript and Html files.
- Run the i18n script automatically, before building the frontend. 

And other ideas?

**Comment by tspmelo** [comment]
@Exotelis What if we have a `messages.en-US.xlf` with no translations inside and never update it.
That way all translations would be missing and the text from the code would be used.
And we would not need to have a specific command/condition for this special case.

Like the following:
```
<xliff xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 http://docs.oasis-open.org/xliff/v1.2/os/xliff-core-1.2-strict.xsd" xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
  <file original="ng2.template" datatype="plaintext" source-language="en-US">
    <body>
    </body>
  </file>
</xliff>
```

**Comment by Exotelis** [comment]
> @Exotelis What if we have a `messages.en-US.xlf` with no translations inside and never update it.
> That way all translations would be missing and the text from the code would be used.
> And we would not need to have a specific command/condition for this special case.
> 
> Like the following:
> 
> ```
> <xliff xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 http://docs.oasis-open.org/xliff/v1.2/os/xliff-core-1.2-strict.xsd" xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
>   <file original="ng2.template" datatype="plaintext" source-language="en-US">
>     <body>
>     </body>
>   </file>
> </xliff>
> ```

That might be a third solution :-) 
As you know a missing `<target>` tag would result in an error when building, a more or less empty file could work.

**Comment by Exotelis** [comment]
Tracked the issue here: https://tracker.ceph.com/issues/42693

**Comment by Exotelis** [comment]
Opened a pr for the bugfix: https://github.com/ceph/ceph/pull/31727

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 1]
```diff
@@ -10,18 +10,19 @@
...
-    "build:pt-BR": "LOCALE=pt-BR npm run build",
-    "build:it-IT": "LOCALE=it-IT npm run build",
-    "build:cs": "LOCALE=cs npm run build",
-    "build:fr-FR": "LOCALE=fr-FR npm run build",
-    "build:id-ID": "LOCALE=id-ID npm run build",
-    "build:ja-JP": "LOCALE=ja-JP npm run build",
-    "build:pl-PL": "LOCALE=pl-PL npm run build",
-    "build:zh-CN": "LOCALE=zh-CN npm run build",
-    "build:zh-TW": "LOCALE=zh-TW npm run build",
+    "build:cs": "LOCALE=cs npm run build-i18n",
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/package.json]: `build-i18n` has been removed, please use `build`
```suggestion
    "build:cs": "LOCALE=cs npm run build",
```

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by Exotelis** [comment]
jenkins test make check

**Pull request merged**